### PR TITLE
Update description of Scheduled Task

### PR DESCRIPTION
**Pull request by RukshiW**

### CODE REVIEW

**Comment by RukshiW** [comment]
Fixes https://github.com/wso2-enterprise/choreo/issues/10625

File: en/docs/scheduled-tasks.md [hunk 1]
```diff
@@ -1,23 +1,73 @@
 # Scheduled Tasks
 
-A scheduled task performs a routine task at scheduled time intervals.
+Learn how to execute routine tasks at specified time intervals via scheduled tasks.
```
Comment (shaniR) [inline en/docs/scheduled-tasks.md:3]: Learn how to use scheduled tasks to execute routing tasks at specified time intervals? -> Wouldn't this be clearer?

File: en/docs/scheduled-tasks.md [hunk 2]
```diff
@@ -1,23 +1,73 @@
...
 
-e.g., You can configure a scheduled task to publish a summary of GitHub issues (i.e., with details such as the number of GitHub issues in each status) as a message in a group chat of a team every day at a specific time.
+## What is a scheduled task?
 
-The in-built low-code and pro-code editors of Choreo allows you to easily configure scheduled tasks.
+A scheduled task performs a routine task at scheduled time intervals. Following are examples of scenarios where you can schedule tasks:
 
-### Develop
+- Logging a session time-out message for a user at a specific time interval (e.g., every 1 hour).
```
Comment (shaniR) [inline en/docs/scheduled-tasks.md:9]: Since they are examples, should we make them generic?

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/scheduled-tasks.md (hunk 1 above)
Comment (RukshiW) [reply en/docs/scheduled-tasks.md]: Changed as suggested.

File: en/docs/scheduled-tasks.md (hunk 2 above)
Comment (RukshiW) [reply en/docs/scheduled-tasks.md]: Removed the time interval given as an example.

File: en/docs/scheduled-tasks.md [hunk 3]
```diff
@@ -1,23 +1,73 @@
...
+Developing a scheduled task is the process of configuring an action to be executed.
 
-You can implement the logic and test it locally in the code server. After that you can commit and push the code changes to the GitHub repository that has been created for your Scheduled Task component, and go back to the Choreo Console to deploy it.
+!!! tip
+The development stage does not require you to specify the time interval at which this action needs to be repeatedly executed. You can specify this when you deploy the Scheduled Task component.
 
-You can also use the connectors supported by Choreo in the scheduled task configuration.
+e.g., If you need to schedule a health check log for an application, you need to configure a scheduled task that gets the heartbeat of the application, and then define the conditions based on which the log indicates that the application is healthy or unhealthy.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/scheduled-tasks.md]: Updated.

File: en/docs/scheduled-tasks.md [hunk 4]
```diff
@@ -1,23 +1,73 @@
...
 
-### Deploy
+### Low-code mode
 
-A scheduled task needs to be deployed so that it can run at the specified time intervals to perform its task. When you deploy a scheduled task, Choreo checks out your latest code, builds it, creates a Docker image, and starts it in an environment managed by Choreo.
+![Scheduled task low-code view](assets/img/scheduled-tasks/scheduled-task-low-code.png){.cInlineImage-full}
 
+The scheduled task shown above checks whether the heartbeat of an application is received via the `HealthCheckAPI` (which is a connector). Then it logs `Healthy` if the application responds with a heartbeat, or `Unhealthy` if there is no response from the application.
+
+The low-code programming model allows developers to use common programming constructs such as loops, conditions(if/else, variable declarations and assignments, logs, data transformations(visual data mapping), and much more in a graphical editor. It also allows developers to connect to any built-in or custom-developed connectors.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/scheduled-tasks.md]: Fixed.

File: en/docs/scheduled-tasks.md [hunk 5]
```diff
@@ -1,23 +1,76 @@
...
+
+
+### Pro-code view
+
+In addition to the low-code view, Choreo provides a pro-code view for implementing the functionality of a scheduled task. You can switch between the two flows at any time. Choreo's online IDE allows viewing the low-code and pro-code editors side-by-side as shown below:
+
+![Scheduled task pro-code view](assets/img/scheduled-tasks/scheduled-task-pro-code.png){.cInlineImage-full}
+
+Choreo gives the developers the flexibility to implement the complete scheduled task using the pro-code view or to use to pro-code view for only some parts of the implementation and then use the low-code view for the rest, based on their preference. Choreo treats the source code of the scheduled task as the single source of truth irrespective of whether it is auto-generated from the low-code implementation or written manually by the developer. Therefore, there are no limitations or restrictions on which view developers may choose to implement the functionality of the scheduled task.
```
Comment (shaniR) [inline en/docs/scheduled-tasks.md:44]: to use to -> to use

**Review by shaniR — APPROVED** [review]

File: en/docs/scheduled-tasks.md (hunk 5 above)
Comment (RukshiW) [reply en/docs/scheduled-tasks.md]: Corrected.

**Pull request merged**